File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    let len = numbers.length;
    function dfs(count,sum){
        if(count===len){
            if(target===sum){
                answer++
            }
            return;
        }
        dfs(count+1,sum+numbers[count])
        dfs(count+1,sum-numbers[count])
    }
    dfs(0,0)
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers,i,n,target):
    result = 0
    if i == len(numbers):
        if n == target:
            return 1
        else:
            return 0
    
    result += dfs(numbers,i+1,n+numbers[i],target)
    result += dfs(numbers,i+1,n-numbers[i],target)
    return result

def solution(numbers, target):
    answer = dfs(numbers,0,0,target)
    return answer